src/domain/Garage.ts

```diff
@@ -0,0 +1,71 @@
+import Car from "./Car.ts";
+
+// ❓ with 적절한 네이밍인가?
+export default class Garage {
+  readonly cars: Car[];
+
+  constructor({ cars = [] }: { cars?: Car[] } = {}) {
+    this.cars = cars;
+  }
+
+  init() {
+    return new Garage({
+      cars: [],
+    });
+  }
+
+  withCar({ name, position }: { name: string; position: number }): Garage {
+    const index = this.cars.findLastIndex((i) => i.name === name);
```

Comment:
이 책임을 자동차한테 보내는거 제안!
```suggestion
    const index = this.cars.findLastIndex(car => car.equal(newCar))
```